agent/internal/detect.go

```diff
@@ -28,8 +31,20 @@ const (
 func (a *agent) detect() error {
 	switch {
 	case a.ArtificialSlots > 0:
+		h := sha256.New()
```

Comment:
neat trick.

Reply:
maybe adding a comment explaining this is to keep slot ids consistent across restarts?

Reply:
idk, or put it in a func with that name, I guess